src/include/miopen/miopen_internal.h

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef GUARD_MIOPEN_INTERNAL_H_
+#define GUARD_MIOPEN_INTERNAL_H_
+
+/* Put experimental APIs here. */
+/* If used, should be included after miopen.h. */
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wextern-c-compat"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Begin of Find Mode API */
+
+/*! @brief Version of the Find Mode API for convolutions.
+*
+* The Find Mode API is experimental and therefore subject to change.
+* This macro allows applications to adapt to the future.
+*/
+#define MIOPEN_API_VERSION_CONVOLUTION_FIND_MODE 1
```

Comment:
This should be a function, as this macro will never be accessible outside of miopen(as this header is not installed).

Reply:
This header is intended to hold APIs that we do not want to expose to the users. However, the contents should be _as if_ it is a public API, by design. This allows us to disclose API at any moment by simple moving text from this file to the public header.

[Notice] I still do not fully agree with reasons of hiding this experimental API (see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/489#discussion_r500558737). The cost of support is zero because we are free to discontinue support of any experimental API at any given moment (provided that we do not forget to #undef the API_VERSION macro).

Reply:
@pfultz2 However please let me know if you need a function (instead of a macro) for your needs.

Reply:
Then we should just remove this macro as no one is going to use it(since its not available to anyone). We are probably just going to check for this function using [`check_symbol_exists`](https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html) in CMake.

Reply:
I do not think so, because I would like to follow the design principle mentioned above:

> This header is intended to hold APIs that we do not want to expose to the users. However, the contents should be as if it is a public API, by design. This allows us to disclose API at any moment by simple moving text from this file to the public header.

Also I do not want to redo the work (even partially) when/if we decide to disclose this API.

Reply:
@pfultz2 
> We are probably just going to check for this function using check_symbol_exists in CMake.

Hmm... AFAICS it checks header files, so it will fail unless we expose the function in the public header.

Reply:
So perhaps you would prefer the approach [shown here](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/489#discussion_r502062038).

Reply:
@pfultz2 I am waiting for feedback from you.

Reply:
But I should still be able to use `check_cxx_source_compiles` in cmake.

Reply:
> Also I do not want to redo the work (even partially) when/if we decide to disclose this API.

I dont think we want to expose the API ever. We should just implement find 2.0.

Reply:
> I dont think we want to expose the API ever. We should just implement find 2.0.

Yes. Unless users ask us to provide them with such a function before 2.0. So who knows?

Reply:
@pfultz2 
> But I should still be able to use `check_cxx_source_compiles` in cmake.

Okay, if this works for you. So I am going to make a C++ function.

Reply:
Done in f9d49b5.